tests/unit/package_managers/test_pip.py

```diff
@@ -3,7 +3,7 @@
 from copy import deepcopy
 from pathlib import Path
 from textwrap import dedent
-from typing import Any, Collection, Literal, Optional, Union
+from typing import Any, Collection, Dict, List, Literal, Optional, Union
```

Comment:
why importing these types and not using dict and list ?

Reply:
Agreed. We're heavily inconsistent in usage of `dict`  vs `Dict` for type hinting, we need to improve on that. 
In context of this commit, this is the only occurrence of `Dict`, so to keep it consistent for now, let's go with `dict`.